### PR TITLE
main activity carousel work

### PR DESCRIPTION
**Pull request by shakedweitz**
* change main carousel to support onClick events
* add soda and fries on select
* add items to beginning of order list

### CODE REVIEW

File: app/src/main/java/com/example/graburger/BurgerItemModel.java [hunk 1]
```diff
@@ -197,23 +248,34 @@ public int getImage(Boolean isUpper ) {
...
+    static SauceType[] all() {
+        return SauceType.values();
+    }
+
+    static SauceType get(int index) {
+        return all()[index];
+    }
 }
 
-public class BurgerItemModel extends FoodItemModel implements Serializable {
```
Comment (Yaelsarusi) [inline app/src/main/java/com/example/graburger/BurgerItemModel.java]: This was needed to serialize the burger and pass it to the intent. What is in line 278 : 'private static final long serialVersionUID = 1L;' is enough?

File: app/src/main/java/com/example/graburger/MainActivity.java [hunk 2]
```diff
@@ -160,13 +150,40 @@ protected void onActivityResult(int requestCode, int resultCode, Intent data) {
             BurgerItemModel orderedBurger = (BurgerItemModel) data.getExtras().getSerializable("order");
             orderedBurger.updateBurger();
             if (requestCode == buildBurgerActivity.CREATE_NEW_ACTIVITY_CODE) {
-                orderList.add(orderedBurger);
+                orderList.add(0, orderedBurger);
```
Comment (Yaelsarusi) [inline app/src/main/java/com/example/graburger/MainActivity.java:153]: Why are you adding the burger in position 0?

File: app/src/main/java/com/example/graburger/MainActivity.java [hunk 3]
```diff
@@ -47,27 +50,13 @@ private void createNewItemCarousel() {
...
-                    intent.putExtra("curBurger", curBurger);
-                    startActivityForResult(intent, buildBurgerActivity.CREATE_NEW_ACTIVITY_CODE);
-                }
-            }
-
-            @Override
-            public void onPageScrollStateChanged(int state) {
-                // TODO : check if we need to implement
+            public boolean onTouch(View view, MotionEvent motionEvent) {
+                gestureDetector.onTouchEvent(motionEvent);
```
Comment (Yaelsarusi) [inline app/src/main/java/com/example/graburger/MainActivity.java:58]: very cool

File: app/src/main/java/com/example/graburger/BurgerItemModel.java (hunk 1 above)
Comment (Yaelsarusi) [inline app/src/main/java/com/example/graburger/BurgerItemModel.java:216]: Yup it's smart to change the FoodItemModel to be serializable. :)

File: app/src/main/java/com/example/graburger/Adapter.java [hunk 4]
```diff
@@ -47,36 +46,24 @@ public boolean isViewFromObject(@NonNull View view, @NonNull Object object) {
...
-//        TextView title, desc;
+        TextView desc;
 
         imageView = view.findViewById(R.id.image);
-//        title = view.findViewById(R.id.title);
-//        desc = view.findViewById(R.id.desc);
+        desc = view.findViewById(R.id.desc);
 
         imageView.setImageResource(models.get(position).getImage());
-//        title.setText(models.get(position).getTitle());
```
Comment (Yaelsarusi) [inline app/src/main/java/com/example/graburger/Adapter.java:61]: All these is the comment is what we need to add text to the orderListCarousel, right?

**Review by Yaelsarusi — APPROVED** [review]

File: app/src/main/java/com/example/graburger/MainActivity.java (hunk 2 above)
Comment (shakedweitz) [reply app/src/main/java/com/example/graburger/MainActivity.java:153]: So the last added item to the order will be shown to the user when added

File: app/src/main/java/com/example/graburger/Adapter.java (hunk 4 above)
Comment (shakedweitz) [reply app/src/main/java/com/example/graburger/Adapter.java:61]: exactly :)

**Pull request merged**